main.rb

```diff
@@ -1,100 +1,138 @@
 require_relative('./app')
 
-def create_student(app)
-  print 'Age: '
-  age = gets.chomp
-  print 'Name: '
-  name = gets.chomp
-  print 'Has parent permission? [Y/N]: '
-  permission = gets.chomp.downcase == 'y'
-  app.create_student(age, name, permission)
-  puts 'Person created successfully'
-end
+class PersonCreator
+  def initialize(app)
+    @app = app
+  end
 
-def create_teacher(app)
-  print 'Age: '
-  age = gets.chomp
-  print 'Name: '
-  name = gets.chomp
-  print 'Specialization: '
-  specialization = gets.chomp
-  app.create_teacher(age, name, specialization)
-  puts 'Person created successfully'
-end
+  def create_person
+    print 'Do you want to create a stduent (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp
+    if choice == '1'
+      create_student
+    elsif choice == '2'
+      create_teacher
+    else
+      puts 'Invalid choice'
+    end
+  end
```

Comment:
Great job team🤝

- [ ] Kindly ensure all requests for input and input validation should've been placed in the `main. rb`. This will make your project conform to the Single Responsibility Rule. Kindly follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more guidance.
- `Good example of Single responsibility rule being followed`
![method-exapmple](https://github.com/BahirHakimy/OOP-school-library/assets/63917725/3cb1bdf9-7c9a-4487-a292-efdad8854611).
- Implement this across all your methods.

Reply:
Hi @1GORDON 
Thanks for your great suggestions,
can you please make your second suggestion a bit more clear?
e.g. in the following method I need the user to select a person from the list of persons or enter its id
```Ruby
def create_rentals
    puts 'Select a book from the following list by number'
    @app.list_all_books(select: true)
    book_idx = gets.chomp.to_i
    puts "\nSelect a person from the following list by number (not id)"
    @app.list_all_people(select: true)
    person_idx = gets.chomp.to_i
    print "\nDate: "
    date = gets.chomp
    @app.create_rental(book_idx, person_idx, date)
    puts 'Rental created successfully'
  end
  ```
  Should I place the statements for getting user input inside `main.rb` or not?